### ISSUE TITLE
45410914 - Rock Band 3

### ISSUE THREAD

**Issue by cod0rama71** [xenia-project/game-compatibility#815]
[Marketplace] http://marketplace.xbox.com/en-GB/Product/Rock-Band-3/66acd000-77fe-1000-9115-d80245410914

Tested on:
https://ci.appveyor.com/project/benvanik/xenia/build/1.0.942-master/job/w9asnpg202gujg6c

# Issues:
Crash after ESRB message (startup)

# Screenshot(s):
https://ibb.co/dNnKU5

# Labels:
state-crash

**Comment by bdeimos** [comment]
Tested on (https://github.com/xenia-project/xenia/commit/94d1106386e571bd371c0cdcb893336f66d364ae)

Crash after intros.

[xenia.zip](https://github.com/xenia-project/game-compatibility/files/2152820/xenia.zip)

**Comment by steve25469** [comment]
Tested on a4a35b4e/Mar 10 2019. Crash after ESRB intro, tested on both master and d3d12 versions. Attached the log file and screen shot of the error message. :)    Great work on this guys, I really look forward to seeing how this develops.
[xenia.log](https://github.com/xenia-project/game-compatibility/files/3022487/xenia.log)
![Desktop Screenshot 2019 03 29 - 06 49 58 82](https://user-images.githubusercontent.com/11917310/55230921-3fbe9900-51ef-11e9-8914-e4a39c87c95b.png)

**Comment by TranzRail** [comment]
Update - 4 August 2019

Game appears to load properly, playing the intros no problem. Colour and framing looks correct. Can navigate game menu and attempt to play a song. Game gets stuck trying to load a song. Very likely due to Xenia not able to decrypt the game data (uses AES). Implementing the related missing modules will most likely get this game and Rock Band 2, Rock Band 3 and others working to some degree. 

Log: https://drive.google.com/file/d/1oqcayUdWlv2HSkG2IpJc7Cr4x3sy_-cL/view?usp=sharing

**Comment by Kramerika** [comment]
I'm not sure which Rock Band game the above post was referring to, but Rock Band 3 still crashes after the Harmonix part of the intro with "the guest has crashed" error.  I cannot get to the game menu, much less load a song.  This is on version c43ccc07/Dec 4 2019.  Log file with crash dump attached.  Looking forward to seeing this game working one day!

[xenia.zip](https://github.com/xenia-project/game-compatibility/files/4035652/xenia.zip)

**Comment by GuidoFida** [comment]
Second on the game still crashing on the intro, saw the state-menus tag but that doesn't seem to be true. On the Feb 16 2020 build.

[xenia.log](https://github.com/xenia-project/game-compatibility/files/4212017/xenia.log)

**Comment by mprachar** [comment]
Hello - wondering if there has been any update here on trying to get RB3 working?

**Comment by ihatecompvir** [comment]
Rock Band 3 has an anti-debugging check that crashes the game if a debugger is present. Stubbing this check out makes the game run, see my video
https://www.youtube.com/watch?v=d2NGnoJs1mQ

It would seem that for Xenia to work with RB3 without requiring patching the game itself, it would have to be made to tell the game that no debugger is present so this check is not triggered.

**Comment by Gliniak** [comment]
@ihatecompvir Thanks for the info and provided address. I will look at it and maybe figure out how to deal with it without patching.

I wouldn't even expect that they can check and crash game if debugger is attached

**Comment by aetha** [comment]
> It isn't related to anti-debugging, but to usage of SEH. Game expect it to fail and rewind itself after MS try-except block

Hi, I'm just wondering if we could expect any updates in the near future? 

What is SEH? Is it a complex issue to work around?

**Comment by Gliniak** [comment]
> 
> 
> > It isn't related to anti-debugging, but to usage of SEH. Game expect it to fail and rewind itself after MS try-except block
> 
> Hi, I'm just wondering if we could expect any updates in the near future?
> 
> What is SEH? Is it a complex issue to work around?

Yes it is quite complicated to fix. For now only solution is to use patch written for canary: https://github.com/xenia-canary/game-patches/blob/main/patches/45410914%20-%20Rock%20Band%203.patch

However iirc game doesn't render stuff correctly and stuck on loading into a gameplay

**Comment by aetha** [comment]
@Gliniak Ah that’s a pity. Thanks for the info though. ❤️

**Comment by JohnSmith2007** [comment]
Sad to see that the game is still having issues on Xenia. Guess I'll have to stick with RPCS3 for now. sad day

**Comment by SteveGrey7** [comment]
I hope to be able to do something with this. Does anyone know where to get the dlc for RB3 PS3 version? Game works better on RPCS3, but I could not find as many dls as on the xbox version((

**Comment by urboiluci** [comment]
rock band the beatles edition works though

**Comment by yugi1408** [comment]
Any update on this game's state yet? I may try running it seems it doesn't seem to have been tested in 2-3 ish years.

**Comment by yugi1408** [comment]
Can confirm the game still does crash on the master branch as of March 25th, 2024.

**Comment by NuclearGeneral** [comment]
Latest build of xenia canary plays rock band 3 past all the intro logos, works in the main menu just fine, but still unable to load in to songs. It just sits at the loading screen and never actually loads into any songs. Be it a song included with the game or dlc song, they do not load. Here is a video of it I took. https://youtu.be/ClCSIPjxjpA

**Comment by Margen67** [comment]
@NuclearGeneral Xenia Canary discussion belongs here: https://github.com/xenia-canary/game-compatibility/issues

**Comment by TomFan-70** [comment]
> > > It isn't related to anti-debugging, but to usage of SEH. Game expect it to fail and rewind itself after MS try-except block
> > 
> > 
> > Hi, I'm just wondering if we could expect any updates in the near future?
> > What is SEH? Is it a complex issue to work around?
> 
> Yes it is quite complicated to fix. For now only solution is to use patch written for canary: https://github.com/xenia-canary/game-patches/blob/main/patches/45410914%20-%20Rock%20Band%203.patch
> 
> However iirc game doesn't render stuff correctly and stuck on loading into a gameplay

Hi thanks for creating the patch.. is it not available anymore? Did i understand correctly the game hangs on loading, even with the patch?
